[macOS Sonoma] Update SPI headers now that -[NSView clipsToBounds] is API

emw-apple opened the pull request:
#### 48d9a42d0a4ce579f1a1180bc1ca45aff0c5a9e4
<pre>
[macOS Sonoma] Update SPI headers now that -[NSView clipsToBounds] is API
<a href="https://bugs.webkit.org/show_bug.cgi?id=258473">https://bugs.webkit.org/show_bug.cgi?id=258473</a>

Reviewed by Alexey Proskuryakov and Jonathan Bedard.

Work towards making the build functional on macOS Sonoma.

* Source/WTF/wtf/PlatformHave.h: Add a HAVE_NSVIEW_CLIPSTOBOUNDS_API
  flag.
* Source/WebCore/PAL/pal/spi/mac/NSViewSPI.h: Only declare clipsToBounds
  on older SDK versions.

Canonical link: <a href="https://commits.webkit.org/265490@main">https://commits.webkit.org/265490@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/42a723b7b37f3e4009fd15888a3b9d27c46f21c1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10999 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11211 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11531 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12646 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/10501 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/11014 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13588 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11193 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13430 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11158 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/12054 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/9271 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13051 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9347 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9942 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/17159 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/9311 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10418 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/10096 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13327 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/10427 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10529 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/8616 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/11106 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9702 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/3031 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2646 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13972 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/11417 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10384 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2799 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Source/WebCore/PAL/pal/spi/mac/NSViewSPI.h:
```diff
@@ -49,7 +49,9 @@ typedef NS_ENUM(NSInteger, NSViewSemanticContext) {
 #endif
 
+#if !HAVE(NSVIEW_CLIPSTOBOUNDS_API)
 @property BOOL clipsToBounds;
```
Any reason this property wasn't guarded before?

emw-apple replied:
clipsToBounds has been SPI for a long time, but was just promoted to API. See https://developer.apple.com/documentation/appkit/nsview/4236466-clipstobounds?language=objc&changes=latest_minor: available since 10.9, but added to the SDK in this release.

aproskuryakov commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1567,3 +1567,7 @@
 #define HAVE_LS_SERVER_CONNECTION_STATUS_RELEASE_NOTIFICATIONS_MASK 1
 #endif
+
+#if PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000
```
Shouldn't this be MAX_ALLOWED? So that we don't unnecessarily complicate building for 10.13 using 10.14 SDK.

emw-apple replied:
Great catch. It's declared in the 14 SDK, so we should be checking SDK version and not deployment target.

emw-apple committed:
[macOS Sonoma] Update SPI headers now that -[NSView clipsToBounds] is API
https://bugs.webkit.org/show_bug.cgi?id=258473

Reviewed by Alexey Proskuryakov and Jonathan Bedard.

Work towards making the build functional on macOS Sonoma.

* Source/WTF/wtf/PlatformHave.h: Add a HAVE_NSVIEW_CLIPSTOBOUNDS_API
  flag.
* Source/WebCore/PAL/pal/spi/mac/NSViewSPI.h: Only declare clipsToBounds
  on older SDK versions.

Canonical link: https://commits.webkit.org/265490@main
